server/src/json-api-deserialize.pipe.ts

```diff
@@ -8,7 +9,11 @@ export class JsonApiDeserializePipe implements PipeTransform {
 
     if (type === 'body') {
       return new Deserializer({
-        keyForAttribute: 'snake_case',
+        keyForAttribute: (attribute) => {
+          if (attribute === 'dcp500-kpluszone')
+            return 'dcp_500kpluszone';
+          return underscore(attribute);
```

Comment:
I looked into the source for jsonapi-serializer to confirm that snake_case defers to this underscore() method